mungegithub/mungers/approvers/owners.go

```diff
@@ -137,15 +137,46 @@ func findMostCoveringApprover(allApprovers []string, reverseMap map[string]sets.
 	return bestPerson
 }
 
+// temporaryUnapprovedFiles returns the list of files that wouldn't be
+// approved by the given set of approvers.
+func (o Owners) temporaryUnapprovedFiles(approvers sets.String) sets.String {
+	ap := NewApprovers(o)
+	for approver := range approvers {
+		ap.AddApprover(approver, "", "")
+	}
+	return ap.UnapprovedFiles()
+}
+
+// KeepMostCoveringApprovers finds who we should keep as suggested approvers given a pre-selection
+// knownApprovers must be a subset of potentialApprovers.
+func (o Owners) KeepMostCoveringApprovers(knownApprovers sets.String, potentialApprovers []string) sets.String {
+	keptApprovers := sets.NewString()
+
+	unapproved := o.temporaryUnapprovedFiles(knownApprovers)
+	reverseMap := o.GetReverseMap()
+
+	for _, suggestedApprover := range o.GetSuggestedApprovers(potentialApprovers).List() {
+		if reverseMap[suggestedApprover].Intersection(unapproved).Len() != 0 {
```

Comment:
>Of the files that are unapproved using a given set of people, you want to choose the the person*s* that covers the largest set right?

Yes. Can you generate a name then? :-)

> it seems weird that the integration between Approvers and Reviewers is so poor. 

I've tried to entertain this idea, but I'm not sure how we can improve the experience. Trying to be smart will make us select the same person for approval and review, and that's the opposite of what we want. It would mean we would only select approvers for reviews ...

> should blunderbuss be more round robin like now that approvers is in place?

Yeah, that might be a good idea. But I think most of this discussion is orthogonal to this issue/PR. And that PR is already a big gain.